### PR TITLE
[FONTVIEW] Hardcoded messages, not available for translation

### PR DESCRIPTION
**Pull request by NikitaP2001**
## Purpose

fontview.c contains several hardcoded elements not available for translation

JIRA issue: [CORE-18952](https://jira.reactos.org/browse/CORE-18952)

## Proposed changes

I have moved such messages in OnInstall procedure to resource file.
- populated message to all language files
- add translation for them ~~generated by ai model~~ in english

### CODE REVIEW

File: base/applications/fontview/fontview.c [hunk 1]
```diff
@@ -93,6 +93,27 @@ ErrorMsgBox(HWND hParent, DWORD dwMessageId, ...)
 	LocalFree(hMemText);
 }
 
+static void
+SuccessMsgBox(HWND hParent, DWORD dwMessageId, ...)
```
Comment (HBelusca) [inline base/applications/fontview/fontview.c:97]: This function could have been merged with the previous one, by adding an extra parameter to that other one for specifying the MB_ICONERROR or MB_ICONINFORMATION.

File: base/applications/fontview/lang/cs-CZ.rc [hunk 2]
```diff
@@ -10,6 +10,12 @@ BEGIN
     IDS_ERROR_NOMEM "K dokončení operace není dostatek paměti."
     IDS_ERROR_NOFONT "Soubor %1 není platným souborem písma."
     IDS_ERROR_NOCLASS "Inicializace okna aplikace selhala."
+    IDS_ERROR_ISINSTALLED "Tento písmo je již nainstalováno!"
+    IDS_ERROR_FONTCPY "Chyba při kopírování souboru písma!"
+    IDS_ERROR_OPENKEY "Chyba při otevírání klíče písem!"
+    IDS_ERROR_REGISTER "Chyba při registraci nového písma!"
+    IDS_SUCCESS "Úspěch"
+    IDS_COMPLETED "Instalace písma dokončena."
```
Comment (HBelusca) [inline base/applications/fontview/lang/cs-CZ.rc:18]: How did you get all these different translations? Do you speak all those 20+ languages? :joy:

**Review by KRosUser2 — APPROVED** [review]
AI translation is usually not welcome... Prefer English by default in case of doubt

File: base/applications/fontview/fontview.c (hunk 1 above)
Comment (NikitaP2001) [reply base/applications/fontview/fontview.c]: fix: add overall FormatMsgBox proc

File: base/applications/fontview/lang/cs-CZ.rc (hunk 2 above)
Comment (NikitaP2001) [reply base/applications/fontview/lang/cs-CZ.rc]: fix: all translation for new added definitions turned to english

**Review by learn-more — APPROVED** [review]

File: base/applications/fontview/lang/bg-BG.rc [hunk 3]
```diff
@@ -10,6 +10,12 @@ BEGIN
     IDS_ERROR_NOMEM "Няма достатъчно място за завършване на действието."
     IDS_ERROR_NOFONT "%1 не е редовен шрифтов файл."
     IDS_ERROR_NOCLASS "Неуспешно изпълнение на класа на прозореца."
+    IDS_ERROR_ISINSTALLED  "This font is already installed!" 
+    IDS_ERROR_FONTCPY "Failed to copy the font file!"
+    IDS_ERROR_OPENKEY "Failed to open the fonts key!"
+    IDS_ERROR_REGISTER "Failed to register the new font!"
+    IDS_SUCCESS "Success"
+    IDS_COMPLETED "Font Installation Completed."
```
Comment (HBelusca) [inline base/applications/fontview/lang/bg-BG.rc:18]: Not sure about the capitalization.
```suggestion
    IDS_COMPLETED "Font installation completed."
```

File: base/applications/fontview/fontview.c [hunk 4]
```diff
@@ -72,27 +72,39 @@ FormatString(
 	return 0;
 }
 
-static void
-ErrorMsgBox(HWND hParent, DWORD dwMessageId, ...)
+static void FormatMsgBox(HWND hParent, DWORD dwMessageId, DWORD dwCaptionId, UINT uType, va_list args)
```
Comment (HBelusca) [inline base/applications/fontview/fontview.c:75]:
```suggestion
static void
FormatMsgBox(
    _In_ HWND hParent,
    _In_ DWORD dwMessageId,
    _In_ DWORD dwCaptionId,
    _In_ UINT uType,
    _In_ va_list args)
```

File: base/applications/fontview/fontview.c [hunk 5]
```diff
@@ -72,27 +72,39 @@ FormatString(
...
+	va_end(args);
+}
+
+static void
+SuccessMsgBox(HWND hParent, DWORD dwMessageId, ...)
+{
+	va_list args;
+	va_start(args, dwMessageId);
+	FormatMsgBox(hParent, dwMessageId, IDS_SUCCESS, MB_ICONINFORMATION, args);
+	va_end(args);
```
Comment (HBelusca) [inline base/applications/fontview/fontview.c:105]: Please use 4-space indentation, instead of tabs.

File: base/applications/fontview/fontview.c [hunk 6]
```diff
@@ -72,27 +72,39 @@ FormatString(
...
-
 	FormatString(FORMAT_MESSAGE_ALLOCATE_BUFFER,
-	              NULL, IDS_ERROR, 0, (LPWSTR)&hMemCaption, 0, NULL);
-
-	MessageBoxW(hParent, hMemText, hMemCaption, MB_ICONERROR);
+	              NULL, dwCaptionId, 0, (LPWSTR)&hMemCaption, 0, NULL);
 
+	MessageBoxW(hParent, hMemText, hMemCaption, uType);
 	LocalFree(hMemCaption);
 	LocalFree(hMemText);
```
Comment (HBelusca) [inline base/applications/fontview/fontview.c:87]: Please fix indentation in that function as well (4-space indentation, instead of tabs).

Comment (binarymaster) [reply base/applications/fontview/fontview.c:93]: This code is already having mixed indentation, I think it would be better to reformat it separately.

**Review by binarymaster — CHANGES_REQUESTED** [review]
@NikitaP2001 it seems you're using anonymous email on GitHub:

image

Please uncheck `Keep my email addresses private` in the [email settings](https://github.com/settings/emails).

File: base/applications/fontview/lang/bg-BG.rc (hunk 3 above)
Comment (NikitaP2001) [reply base/applications/fontview/lang/bg-BG.rc]: fix: change capitalization, also for all language files

File: base/applications/fontview/fontview.c (hunk 6 above)
Comment (binarymaster) [reply base/applications/fontview/fontview.c]: I have committed formatting fixes into master separately, and resolved it there. Should be good now.

**Review by HBelusca — APPROVED** [review]

**Pull request merged**